### DISCUSSION TITLE
Usb adapter for wii u, switch and pc

### DISCUSSION THREAD

**Discussion by lasse333** [darthcloud/BlueRetro#594] [Ideas]
Is it possible to use the "GameCube BlueRetro Internal Adapter" board to mod a gamecube adapter or perhaps make something similar to the usb gamecube adapter using the blueretro tech i would personally love that

**Comment by Sowden** [comment]
Hey there.  Unfortunately, the only game that works with the Wii U Gamecube USB adapter is SmashBros because it's programed for it.  No other game takes that type of input it gives, so a Blueretro won't work with a Wii U that way.  Now in theory, although I'm not aware of anyone testing this, the Wii-Ext that works with NES Classic has the same port that the Wiimote has.  My theory is that if you connect a Wiimote to the Wii U, then plug a Blueretro into the Wiimote, then it should read the Blueretro like it's a Wii Pro Controller.  But you can use a Wii U Pro Controller in its place or anything else really.  It's a bit janky, but it should get bluetooth devices to work in Wii U and more importantly vWii.  Which it should then work with homebrew and emulators.  But again, all in theory.  I'm planning on testing it, but it's gonna take me a while before I get it.

**Reply by lasse333** [reply]
It was more to use for the switch i would love a usb game cube adapter that could do the same as the real gamecube with the lazerbear board

**Reply by Sowden** [reply]
Oh, well if your looking on the switch then you have plenty of options for you.  The 8bitdo Gbros comes to mind.

https://stoneagegamer.com/8bitdo-gbros-adapter-for-nintendo-switch-works-with-wired-gamecube-classic-edition-controllers.html

**Reply by lasse333** [reply]
What i was thinking was this
![GameCube_20BlueRetro_20Internal_20Adapter_900x](https://user-images.githubusercontent.com/20445208/216664922-9179bd82-07d6-403d-92c3-d2a383f2c3fd.png)

Inside one of these if its possible
![nintendo-gamecube-controller-adapter](https://user-images.githubusercontent.com/20445208/216665031-fc9bdb38-f38e-4ea4-8106-fbba29dc1e93.jpg)

**Reply by Sowden** [reply]
Oh man I dono.  First off the top of my head someone would need to 3d print a new case.  But then you'd have to figure out how to power it.  But what features out if the lazerbear board would you want for the Switch?  Just curious.

**Reply by lasse333** [reply]
I like the leds in the ports and the fact that either its Bluetooth or gamecube controller (gamecube controller has higher priority) but it would be cool if it was a usb adapter for wii u and switch as well and potentially pc if you have the right software

**Comment by kleffa** [comment]
Since wiiiu have bloopair, a homebrew, blueretro is not need in that
For ps3, pc or switch it may be a good idea

**Reply by lasse333** [reply]
I didn't know about bloopair before but it would be cool to see a usb game cube controller adapter with the built-in blue retro and added hid interface for pc

**Reply by kleffa** [reply]
Please correct me, but is GameCube support 4 controllers, but not all buttons, PS instead support 2 controllers, but 2 buttons more and is more standar, 
Which controller have more buttons of any blueretro supported?

**Reply by lasse333** [reply]
The ps4 controller has the most native buttons besides the ps2 did support 4 controllers but you needed an extra accessory for that and yes the gamecube does not have a left bumber and the select/back/menu button (the one next to the start button)

**Reply by kleffa** [reply]
Well, I think it should work, but for pc as wiiiu, I think a simple Bluetooth dongle and some software configuration will work better than a adapter for a adapter.
Nintendo switch and PS3, I guess are more interesting for this, also think a adapter could be done with another module, even integrate in case os esp32 s3 (sadly Xbox 360, one, series, you need a extra adapter)

**Reply by Sowden** [reply]
Yes thats true, Bloopair does work great using Retroarch through Wii U homebrew.  But it stops anywhere past vWii.  And the greatest N64 and PS1 (which has been exploding over the past five months or so) are still only in Wii homebrew.  So Bloopair doesn't help in that situation.  I am working right now with the Retro Rosetta right now for my Atari computers.  But I will buy another for the Wii-Ext.  Which if it should work right, will work through the bottom port on the Wiimote.  We will see, but I will report back when I get to it.
